plugins/inputs/snmp_trap/snmp_trap.go

```diff
@@ -0,0 +1,147 @@
+package snmp_trap
+
+import (
+	"net"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/influxdata/telegraf/plugins/inputs/snmp"
+
+	"github.com/soniah/gosnmp"
+)
+
+type SnmpTrap struct {
+	ServiceAddress string `toml:"service_address"`
+
+	acc      telegraf.Accumulator
+	listener *gosnmp.TrapListener
+	wg       sync.WaitGroup
+	timeFunc func() time.Time
+
+	makeHandlerWrapper func(func(packet *gosnmp.SnmpPacket, addr *net.UDPAddr)) func(packet *gosnmp.SnmpPacket, addr *net.UDPAddr)
+
+	Log telegraf.Logger
+}
+
+var sampleConfig = `
+  ## Local address and port to listen on.  Omit address to listen on
+  ## all interfaces.  Example "127.0.0.1:1234", default ":162"
+  #service_address = :162
+`
+
+func (s *SnmpTrap) SampleConfig() string {
+	return sampleConfig
+}
+
+func (s *SnmpTrap) Description() string {
+	return "Receive SNMP traps"
+}
+
+func (s *SnmpTrap) Gather(_ telegraf.Accumulator) error {
+	return nil
+}
+
+func init() {
+	inputs.Add("snmp_trap", func() telegraf.Input {
+		return &SnmpTrap{
+			timeFunc:       time.Now,
+			ServiceAddress: ":162",
+		}
+	})
+}
+
+func (s *SnmpTrap) Init() error {
+	return nil
+}
+
+func (s *SnmpTrap) Start(acc telegraf.Accumulator) error {
+	s.acc = acc
+	s.listener = gosnmp.NewTrapListener()
+	s.listener.OnNewTrap = makeTrapHandler(s)
+	s.listener.Params = gosnmp.Default
+
+	// wrap the handler, used in unit tests
+	if nil != s.makeHandlerWrapper {
+		s.listener.OnNewTrap = s.makeHandlerWrapper(s.listener.OnNewTrap)
+	}
+
+	s.wg.Add(1)
+	go func() {
+		defer s.wg.Done()
+
+		// no ip means listen on all interfaces, ipv4 and ipv6
+		err := s.listener.Listen(s.ServiceAddress)
+		if err != nil {
+			s.Log.Errorf("error in listen: %s", err)
+		}
+	}()
+
+	<-s.listener.Listening()
```

Comment:
Add a info log message with the listening address and port, keeping style with **most** of the other listening plugins.  For example:
```
2019-11-15T00:04:11Z I! [inputs.socket_listener] Listening on udp://0.0.0.0:8094
```